### PR TITLE
add LayerNorm

### PR DESCRIPTION
**Pull request by Hejunchao6**

### CODE REVIEW

File: src/Nncase.Tests/Evaluator/UnitTestEvaluatorNN.cs [hunk 1]
```diff
@@ -266,6 +267,24 @@ public void TestHardmax()
...
+        var b = OrtKI.Random(new[] { shape[^1] });
+        var axis = 1L;
+        var epsilon = 1e-05f;
+
+        // var expect = OrtKI.LayerNormalization(x, scale, b, axis, epsilon, 1L);
+        var expr = IR.F.NN.LayerNorm((int)axis, epsilon, x.ToTensor(), scale.ToTensor(), b.ToTensor());
+        CompilerServices.InferenceType(expr);
+        expr.Evaluate().AsTensor();
+
+        // Assert.Equal(expect, expr.Evaluate().AsTensor());
```
Comment (xhuohai) [inline src/Nncase.Tests/Evaluator/UnitTestEvaluatorNN.cs:285]: need to open test

File: src/Nncase.Transform/Rules/Neutral/FoldLayerNorm.cs [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+using Nncase.IR;
+using Nncase.IR.Math;
+using Nncase.PatternMatch;
+using static Nncase.IR.F.NN;
+using static Nncase.PatternMatch.F.Math;
+using static Nncase.PatternMatch.F.Tensors;
+using static Nncase.PatternMatch.Utility;
+
+namespace Nncase.Transform.Rules.Neutral;
```
Comment (xhuohai) [inline src/Nncase.Transform/Rules/Neutral/FoldLayerNorm.cs:12]: need to add test for each pattern

**Review by xhuohai — CHANGES_REQUESTED** [review]

**Comment by Hejunchao6** [comment]
add layer_norm&&fold_layer_norm

File: src/Native/src/kernels/stackvm/reference/layernorm.cpp [hunk 3]
```diff
@@ -0,0 +1,75 @@
+/* Copyright 2019-2021 Canaan Inc.
+ *
```
Comment (xhuohai) [inline src/Native/src/kernels/stackvm/reference/layernorm.cpp:2]: conflict with layer_norm.cpp

File: src/Nncase.Tests/Evaluator/UnitTestEvaluatorNN.cs [hunk 4]
```diff
@@ -266,6 +267,23 @@ public void TestHardmax()
         DoHardmax(ortTensor, nncaseTensor, 1L);
     }
 
+    [Fact]
+    public void TestLayerNorm()
+    {
```
Comment (xhuohai) [inline src/Nncase.Tests/Evaluator/UnitTestEvaluatorNN.cs:272]: need to test more parameters, especially axis

**Review by xhuohai — CHANGES_REQUESTED** [review]

File: src/Nncase.Tests/Evaluator/UnitTestEvaluatorNN.cs [hunk 5]
```diff
@@ -266,6 +267,41 @@ public void TestHardmax()
...
+        var epsilon = 1e-05f;
+        {
+            var shape = new long[] { 1, 3, 16, 16 };
+            var x = OrtKI.Random(shape);
+            for (int i = 0; i < shape.Length; i++)
+            {
+                var scale = OrtKI.Random(new[] { shape[i] });
+                var b = OrtKI.Random(new[] { shape[i] });
+                var axis = i;
```
Comment (xhuohai) [inline src/Nncase.Tests/Evaluator/UnitTestEvaluatorNN.cs:281]: negative axis is missing

**Review by xhuohai — CHANGES_REQUESTED** [review]

**Pull request merged**